src/Core/WinSWCore/Native/Advapi32.cs

```diff
@@ -79,6 +79,26 @@ public void ChangeConfig(TimeSpan failureResetPeriod, List<SC_ACTION> actions)
             }
         }
 
+        /// <summary>
+        /// Sets the DelayedAutoStart flag.
+        /// It will be applioed to services with Automatic startup mode only.
+        /// If the platform does not support this flag, an exception may be thrown.
+        /// </summary>
+        /// <param name="enabled">Value to set</param>
+        /// <exception cref="Exception">Operation failure, e.g. the OS does not support this flag</exception>
+        public void SetDelayedAutoStart(bool enabled)
+        {
+            SERVICE_DELAYED_AUTO_START settings = new SERVICE_DELAYED_AUTO_START
+            {
+                fDelayedAutostart = enabled
+            };
+
+            if (!Advapi32.ChangeServiceConfig2(Handle, SERVICE_CONFIG_INFOLEVEL.SERVICE_CONFIG_DELAYED_AUTO_START_INFO, ref settings))
+            {
+                throw new Exception("Failed to change the DelayedAutoStart setting", new Win32Exception());

```

Comment:
Empty constructor captures in automatically: https://msdn.microsoft.com/en-us/library/asy2azk0(v=vs.110).aspx